### PR TITLE
Cut a release branch whenever a #major# is accepted

### PR DESCRIPTION
**Pull request by EWhite613**
#FEATURE#

Add feature for when a major pr is accepted, create a new remote branch (That matches the version)

### CODE REVIEW

**Comment by job13er** [comment]
Nice feature idea but I'm not really sure of it's placement or usefulness. A new tag is already created for every bump, a branch can easily be created from any of those tags. Why do we want/need this? If there is really a need for it, I think there's a better place for where it fits in. I'll take a closer look when I'm at the computer.

**Comment by sglanzer** [comment]
I asked for this against majors because it makes it clean to maintain prior releases with a pre-cut branch.  We've had cases already where we should have cut a branch, but didn't and then we're looking back through tags trying to find the right point to branch from.  This removes the guesswork from my perspective

**Comment by job13er** [comment]
I'm particularly concerned because people will continue working on `master` even after the release branch is created, so version `2.0.1` will likely be off of `master`, and the `2.0.0` branch will be out of date. Generally, the workflow I was considering was that when a maintenance branch was necessary, it would be created based on the last release on the previous major, after `master` had already moved on. I suppose we could do that automatically, and maybe that was the intended use of this, but from the first look, it looked like a `2.0.0` branch would be created as soon as that version was bumped.

**Comment by sglanzer** [comment]
That's a good point - you want to cut the maintenance branch for the prior, not a branch for the current

**Comment by sglanzer** [comment]
So are we good with this if it cuts the maintenance branch (tagged as the prior major) on accepting a new major?

**Comment by job13er** [comment]
> We've had cases already where we should have cut a branch, but didn't and then we're looking back through tags trying to find the right point to branch from. This removes the guesswork from my perspective

I'm not sure where the guesswork is, the tags are in order and it's the one right before the major, but I see no harm in doing it automatically on the version prior to the new major. I'm doing quite a bit of refactoring in my bitbucket branch and am already realizing I need to separate out the stuff that deals with git commands and pushes into a `ci` abstraction b/c we need to handle Travis differently than TeamCity, just like I did with `vcs` for github vs. bitbucket. I'll get this functionality into that branch too unless there's any objection.

File: lib/bumper.js [hunk 1]
```diff
@@ -45,11 +45,13 @@ class Bumper {
   bump () {
     const vcs = this.vcs
     const config = this.config
+    let PrInfo = null
```
Comment (job13er) [inline lib/bumper.js]: just to put it out there, what's with the `PascalCase` variable here?

File: lib/utils.js [hunk 2]
```diff
@@ -283,6 +283,25 @@ const utils = {
...
+      // use info.version
+      exec(`git checkout -b  ${info.version}`, function (error, stdout, stderr) {
+        if (error !== null) {
+          console.log(`exec error: ${error}`)
+        } else {
+          exec(`git push origin ${info.version}`)
+        }
+      })
+    }
+    return info
```
Comment (job13er) [inline lib/utils.js]: There's a couple problems I see here:

 1. `createReleaseBranch()` doesn't actually always create a release branch, so the function might be better named `maybeCreateReleaseBranch()` or something to that effect. 
 1. As mentioned in the general comment, you're creating a branch off of the new change, what you really want is a branch created from the most recent tag *before* this tag gets made. I can incorporate this feature into the bitbucket branch I'm working on.

**Comment by job13er** [comment]
I'll incorporate this into #36

**Pull request closed without merging**